packages/core/src/shipping/strategies/braintree/braintree-accelerated-checkout-shipping-strategy.ts

```diff
@@ -0,0 +1,219 @@
+    ): Promise<InternalCheckoutSelectors> {
+        return this.store.dispatch(this.consignmentActionCreator.updateAddress(address, options));
+    }
+
+    selectOption(
+        optionId: string,
+        options?: ShippingRequestOptions,
+    ): Promise<InternalCheckoutSelectors> {
+        return this.store.dispatch(
+            this.consignmentActionCreator.selectShippingOption(optionId, options),
+        );
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        return Promise.resolve(this.store.getState());
+    }
+
+    async initialize(
+        options: BraintreeAcceleratedCheckoutInitializeOptions,
+    ): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options || {};
+
+        if (!methodId) {
+            throw new InvalidArgumentError(
+                'Unable to proceed because "methodId" argument is not provided.',
+            );
+        }
+
+        if (!this.shouldRunAuthenticationFlow()) {
+            return Promise.resolve(this.store.getState());
+        }
+
+        try {
+            await this.store.dispatch(this.paymentMethodActionCreator.loadPaymentMethod(methodId));
+            await this.runAuthenticationFlowOrThrow(methodId);
+        } catch (error) {
+            // Info: we should not throw any error here to avoid
+            // customer stuck on shipping step due to the payment provider
+            // custom flow
+        }
+
+        return Promise.resolve(this.store.getState());
+    }
+
+    private shouldRunAuthenticationFlow(): boolean {
+        const state = this.store.getState();
+        const cartId = state.cart.getCart()?.id;
+        const payPalConnectSessionId = this.browserStorage.getItem('sessionId');
+        const paymentProviderCustomer = state.paymentProviderCustomer.getPaymentProviderCustomer();
+
+        return !paymentProviderCustomer?.authenticationState && payPalConnectSessionId === cartId;
+    }
+
+    private async runAuthenticationFlowOrThrow(methodId: string) {
+        const state = this.store.getState();
+        const cart = state.cart.getCartOrThrow();
+        const countries = state.countries.getCountries() || [];
+        const customer = state.customer.getCustomer();
+        const billingAddress = state.billingAddress.getBillingAddress();
+        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
+        const { clientToken, initializationData } = paymentMethod;
+
+        if (!clientToken || !initializationData) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        this.braintreeIntegrationService.initialize(clientToken, initializationData);
+
+        const braintreeConnect = await this.braintreeIntegrationService.getBraintreeConnect(
+            cart?.id,
+        );
+
+        const customerEmail = customer?.email || billingAddress?.email;
+
+        if (!customerEmail) {
+            return Promise.resolve(state);
```

Comment:
Actually, we don't need to return anything here and in the code below in this function